Sprint 9

IlyaKorastylev opened the pull request.

IlyaKorastylev committed:
В SettingsActivity.kt добавлены обработчики нажатий на кнопки Поделиться, Поддержка, и Пользовательское соглашение.

IlyaKorastylev committed:
В SettingsActivity.kt переопределены методы onSaveInstanceState и onRestoreInstanceState. В переменной savedSearchText сохраняются данные, введенные в EditText

faritowich commented on app/src/main/java/com/example/playlistmaker/SettingsActivity.kt:
```diff
@@ -1,21 +1,54 @@
+        val shareButton = findViewById<Button>(R.id.share_button)
+
+        shareButton.setOnClickListener {
+            val message = "https://practicum.yandex.ru/android-developer/"
```
Стоит вынести в ресурсы

faritowich commented on app/src/main/java/com/example/playlistmaker/SettingsActivity.kt:
```diff
@@ -1,21 +1,54 @@
+
+        supportButton.setOnClickListener {
+            val subject = "Сообщение разработчикам и разработчицам приложения Playlist Maker"
+            val message = "Спасибо разработчикам и разработчицам за крутое приложение!"
```
Стоит вынести в ресурсы

faritowich commented on app/src/main/java/com/example/playlistmaker/SettingsActivity.kt:
```diff
@@ -1,21 +1,54 @@
+            supportIntent.putExtra(Intent.EXTRA_EMAIL, arrayOf("i.korastylev@yandex.ru"))
+            supportIntent.putExtra(Intent.EXTRA_SUBJECT, subject)
+            supportIntent.putExtra(Intent.EXTRA_TEXT, message)
+            startActivity(supportIntent)
```
Можно ещё использовать котлиновские функции области видимости, чтобы такие фрагменты кода сокращать до таких:

```

            Intent(Intent.ACTION_SENDTO).apply {
                val subject = "Сообщение разработчикам и разработчицам приложения Playlist Maker"
                val message = "Спасибо разработчикам и разработчицам за крутое приложение!"
                data = Uri.parse("mailto:")
                putExtra(Intent.EXTRA_EMAIL, arrayOf("i.korastylev@yandex.ru"))
                putExtra(Intent.EXTRA_SUBJECT, subject)
                putExtra(Intent.EXTRA_TEXT, message)
                startActivity(this)
            }


```
Они иногда удобны, иногда нет, с практикой придёт.
Про такие методы почитать можно здесь:
https://kotlinlang.ru/docs/reference/scope-functions.html

faritowich commented on app/src/main/java/com/example/playlistmaker/SettingsActivity.kt:
```diff
@@ -1,21 +1,54 @@
+        settingsBackButton.setOnClickListener {
+            val settingsBackIntent = Intent(this, MainActivity::class.java)
+            startActivity(settingsBackIntent)
             finish()
```
Здесь не нужны эти строки:
```
            val settingsBackIntent = Intent(this, MainActivity::class.java)
            startActivity(settingsBackIntent)
```
Вызов метода finish() и так закрывает текущую активити и открывает предыдущую

faritowich commented on app/src/main/java/com/example/playlistmaker/MyApplication.kt:
```diff
@@ -0,0 +1,24 @@
+    }
+
+    companion object {
+        const val SEARCH_LINE = "SEARCH_LINE"
```
Не нужно эту константу выносить в Application, она имеет локальное значение только для SearchActivity, там её и стоит объявлять

faritowich commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -1,9 +1,83 @@
+            android:layout_height="wrap_content"
+            android:src="@drawable/search14"
+            android:layout_gravity="center_vertical"
+            android:layout_marginStart="13dp"
```
Обычно рекомендуется делать размеры кратными 8 (или 4). Это обеспечивает единообразие дизайна, а также связано с тем, что большинство размеров экрана делятся на 8. Иногда эту рекомендацию приходится нарушать, но лучше по возможности ей следовать.
Также см. https://material.io/design/layout/spacing-methods.html#baseline-grid

faritowich commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -1,9 +1,83 @@
+        android:orientation="horizontal"
+        android:background="@drawable/rounded_edittext"
+        android:layout_marginStart="@dimen/search_margin_start_end"
+        android:layout_marginEnd="@dimen/search_margin_start_end"
```
Можно объединить в layout_marginHorizontal

faritowich commented on app/src/main/res/values-night/styles.xml:
```diff
@@ -0,0 +1,38 @@
+        <item name="android:textColor">@color/maket_black</item>
+        <item name="android:layout_marginTop">@dimen/main_margin_top_button</item>
+        <item name="android:layout_marginStart">@dimen/main_buttons_left_margin</item>
+        <item name="android:layout_marginEnd">@dimen/main_buttons_right_margin</item>
```
Лучше отступы, высоту и ширину не выносить в стиль. Если на экране будет много элементов, будет сложнее ориентироваться с их взаимным расположением, когда отвечающие за это атрибуты разнесены по разным файлам - что-то может быть в лейауте, что-то в стиле. В стилях обычно содержат атрибуты, отвечающие за внешний вид вьюхи - цвета, шрифты, фоны и т.д.

IlyaKorastylev committed:
1. В SettingsActivity.kt тексты сообщений и заголовок письма в техподдержку перенесены в строковые ресурсы.
2. Скорректирован shareButton.setOnClickListener для корректного вызова share-диалога.
3. В из стилей вынесены отступы.
4. В SettingsActivity.kt и SearchActivity.kt скорректированны обработчики возврата на главный экран.
5. Константа SEARCH_LINE перенесена в SearchActivity.kt
